tests/integration/deposits/test_get_deposits.py

```diff
@@ -570,6 +570,280 @@ def test_get_deposits_with_facets_get_types_and_versions(
+    assert resp.json['hits']['hits'][1]['id'] == deposit_pid_2
+    assert resp.json['hits']['hits'][2]['id'] == deposit_pid_1
+
+    # Test with other records and different schema
+    create_schema(
+        'test-ana',
+        experiment='ATLAS',
+        deposit_mapping=deposit_mapping_2,
+        deposit_schema={
+            'type': 'object',
+            'properties': {
+                "stat": {
+                    "type": "object",
+                    "properties": {
+                        "main_status": {
+                            "type": "string",
+                            "enum": [
+                                "Preliminary results",
+                                "Final results"
+                            ]
+                        },
+                        "sub_status": {
+                            "type": "string",
+                            "title": "Sub stage",
+                            "enum": [
+                                "Planned",
+                                "Conference note under review",
+                                "Paper published"
+                            ]
+                        }
+                    },
+                }
+            }
+        },
+    )
+    deposit_pid_4 = create_deposit(owner, 'test-ana')['_deposit']['id']
+
+    # total should be 4
+    resp = client.get(
+        '/deposits/',
+        headers=auth_headers_for_user(owner) + [('Accept', 'application/basic+json')],
+    )
+    assert resp.json['hits']['total'] == 4
+
+    client.put('/deposits/{}'.format(deposit_pid_4),
+        headers=headers + json_headers,
+        data=json.dumps({
+            "stat": {
+                "main_status": "Preliminary results",
+                "sub_status": "Conference note under review"
+        }}))
+    time.sleep(5)
+
+    # Test asc order
+    resp = client.get('/deposits?q=&sort=stage',headers=headers+json_headers)
+
+    assert resp.status_code == 200
+    assert resp.json['hits']['hits'][0]['id'] == deposit_pid_1
+    assert resp.json['hits']['hits'][1]['id'] == deposit_pid_2
+    assert resp.json['hits']['hits'][2]['id'] == deposit_pid_4
+    assert resp.json['hits']['hits'][3]['id'] == deposit_pid_3
+
+    # Test without sub_status
+    deposit_pid_5 = create_deposit(owner, 'test-ana')['_deposit']['id']
+
+    # total should be 5
+    resp = client.get(
+        '/deposits/',
+        headers=auth_headers_for_user(owner) + [('Accept', 'application/basic+json')],
+    )
+    assert resp.json['hits']['total'] == 5
+
+    client.put('/deposits/{}'.format(deposit_pid_5),
+        headers=headers + json_headers,
+        data=json.dumps({
+            "stat": {
+                "main_status": "Preliminary results"
+        }}))
+    time.sleep(5)
+
+    resp = client.get('/deposits?q=&sort=stage',headers=headers+json_headers)
+    assert resp.status_code == 200
+    assert resp.json['hits']['total'] == 4
+
```

Comment:
lets add a 3rd schema that is random, without any `analysis_stage`, to check that everything works
- there should be deposits from all schemas
- do `/deposits` query to get all
- do sort query to check that it works and no errors come back